### PR TITLE
Django 3.2 upgrade initial CI run

### PR DESCRIPTION
**Pull request by briddo**
### Fixes: #1277

### CODE REVIEW

File: nautobot/core/apps/__init__.py [hunk 1]
```diff
@@ -652,3 +652,13 @@ class NautobotServerConfig(ConstanceConfig):
     """Override "Constance" app name to "Configuration"."""
 
     verbose_name = "Configuration"
+
+
+class ConstanceDatabaseAppConfig(AppConfig):
```
Comment (glennmatthews) [inline nautobot/core/apps/__init__.py:657]: Should this inherit from `constance.apps.ConstanceConfig` instead? There's some default behavior there (https://github.com/jazzband/django-constance/blob/master/constance/apps.py) that I'm not sure but what we may want to still retain.

**Review by glennmatthews — COMMENTED** [review]
We don't have a ton of test coverage for the Constance integration - could you do some manual testing to at least make sure that `/admin/constance/config/` loads correctly and you can set values there that persist?

File: nautobot/core/apps/__init__.py (hunk 1 above)
Comment (briddo) [reply nautobot/core/apps/__init__.py:657]: No I don't so as the issue is with `constance.backends.database` not `constance`

**Comment by briddo** [comment]
> We don't have a ton of test coverage for the Constance integration - could you do some manual testing to at least make sure that `/admin/constance/config/` loads correctly and you can set values there that persist?

Just tested by setting HIDE_RESTRICTED_UI=True. Logged out and back in as a restricted user and verified UI was restricted. Then logged out and back in as superuser and set HIDE_RESTRICTED_UI=False and logged out and back in as a restricted user and verified UI was not restricted.

File: nautobot/core/apps/__init__.py (hunk 1 above)
Comment (briddo) [reply nautobot/core/apps/__init__.py:657]: Further to this, we already extend `constance.apps.ConstanceConfig` here: `nautobot.core.apps.NautobotServerConfig`. This is separate from `constance.backends.database` though.

Comment (glennmatthews) [reply nautobot/core/apps/__init__.py:657]: Good catch, I'd forgotten that these were two separate apps.

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**